cv_next/app/feed/components/CVItemRSC.tsx

```diff
@@ -1,61 +1,117 @@
 "use client";
+import { getBlurredCv } from "@/app/actions/cvs/getBlurCv";
 import Image from "next/image";
 import Link from "next/link";
-import generateImageUrl from "@/helpers/imageURLHelper";
 import Categories from "@/types/models/categories";
-import { useEffect, useState, cache } from "react";
-import { getBlurredCv } from "@/app/actions/cvs/getBlurCv";
+import { useState, useEffect, cache } from "react";
+import { getIdFromLink, getGoogleImageUrl } from "@/helpers/imageURLHelper";
+import {
+  getImageURL,
+  revalidatePreview,
+  getUserName,
+} from "@/app/actions/cvs/preview";
+import Definitions from "@/lib/definitions";
 
-const getBlur = cache(async (imageURL: string) => {
-  return await getBlurredCv(imageURL);
-});
 interface CVCardProps {
   cv: CvModel;
 }
 
+const getBlur = cache(async (imageURL: string) => {
+  return await getBlurredCv(imageURL);
+});
+
+const getURL = cache(async (cvId: string) => {
+  return await getImageURL(cvId);
+});
+
+const getCachedUserName = cache(async (userId: string) => {
+  return await getUserName(userId);
+});
+
 export default function CVItemRSC({ cv }: CVCardProps) {
+  const [realURL, setRealURL] = useState("");
+  const [authorName, setAuthorName] = useState("");
+
   const [base64Data, setBase64Data] = useState(
-    "data:iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7p5fDwAFlAI2LB7hbAAAAABJRU5ErkJggg=="
+    "data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7p5fDwAFlAI2LB7hbAAAAABJRU5ErkJggg=="
   );
 
   const categoryLink = `/feed?category=${Categories.category[
     cv.category_id
   ].toLowerCase()}`;
-  const imageUrl = generateImageUrl(cv.document_link);
-  const formattedDate = new Date(cv.created_at).toLocaleDateString("en-US");
 
   useEffect(() => {
-    (async () => {
-      const base64 = await getBlur(imageUrl);
+    const revalidateImage = async () => {
+      await revalidatePreview(cv.document_link);
+      const imageUrl =
+        (await getURL(getIdFromLink(cv.document_link) || "")) ?? "";
+      setRealURL(imageUrl);
+    };
+    const getAuthorName = async () => {
+      const userUploading = (await getCachedUserName(cv.user_id || "")) || "";
+      setAuthorName(userUploading);
+    };
+    const getBlurCv = async () => {
+      const base64 = await getBlur(getGoogleImageUrl(cv.document_link));
       setBase64Data(base64);
-    })();
-  }, [imageUrl]);
+    };
+    getBlurCv();
+    getAuthorName();
+    revalidateImage(); // Immediately invoke revalidateImage
```

Comment:
FYI, if you don't `await` on each of these functions you're calling them all asynchronously, i.e. they don't depend on each other. Just making sure because I don't understand the comment.

Reply:
Technically yeah this comment is redundant but these should really depend on each other(maybe only the getblurcv and the revalidate because they both fetch the entry from google but they both also cache it so not really sure if it matters which one runs first)

Reply:
Well, if you want to make sure e.g. getBlurCv is ran before revalidateImage you should `await getBlurCv()`..